src/AddressUkPafLookup.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+
+namespace Gcd\UkAddressLookup;
+
+
+use Rhubarb\Leaf\Presenters\Controls\CompositeControlPresenter;
+
+class AddressUkPafLookup extends CompositeControlPresenter
+{
+    private $defaultValues;
+    protected $view;
+    const pafServerUrl = "http://paf.gcdtech.com/paf-data.php?simple=1&api=2&output=json";
+
+    public function __construct($name = "")
+    {
+        parent::__construct($name);
+    }
+
+    protected function initialiseModel()
+    {
+        parent::initialiseModel();
+
+        $this->model->Country = "GB";
+    }
+
+    protected function createView()
+    {
+        $view = new AddressUkPafLookupView();
+        return $view;
+    }
+
+    protected function configureView()
+    {
+        parent::configureView();
+        $this->view->defaultValues = $this->defaultValues;
+
+        $this->view->AttachEventHandler( "SearchPressed", function ( $houseNumber, $postCodeSearch ) {
+            if(!isset($postCodeSearch)) {
+                return json_decode([]);
+            }
+            $searchParams = [];
+            $searchParams[ 'postcode' ] = urlencode( $postCodeSearch );
+
+            $pafSettings = new PafSettings();
+            $searchParams[ 'apikey' ] = $pafSettings->ApiKey;
+
+            if (isset( $houseNumber )) {
+                $searchParams[ 'num' ] = urlencode( $houseNumber );
+            }
+            $requestUrl = implode( '&', [ self::pafServerUrl, http_build_query( $searchParams, '&' ) ] );
```

Comment:
That's an odd way of concatenating 2 strings. Why not just have `self::pafServerUrl . '&' . http_build_query(...`?

Also, the 2nd parameter to http_build_query() is a numeric prefix, not the argument separator. What's more, you don't need to specify the argument separator, as it will default to '&'.